src/main/java/org/folio/innreach/domain/service/impl/InnReachTransactionActionServiceImpl.java

```diff
@@ -316,6 +332,34 @@ public void transferItemHold(UUID transactionId, String itemBarcode) {
     eventPublisher.publishEvent(MoveRequestEvent.of(transaction, item));
   }
 
+  private void recallItem(InnReachTransaction transaction, UUID itemId) {
+    List<RequestDTO> requestList = requestService.getRequestsByItemId(itemId);
+    for (RequestDTO request : requestList) {
+      if (request.getStatus() == OPEN_NOT_YET_FILLED && request.getRequestType() == RequestType.RECALL.toString()) {
+
+        updateTransactionOnLoanRecallRequested(transaction);
+
+      } else {
+
+        var recallUser = recallUserService.getRecallUserForCentralServer(transaction.getCentralServerCode());
+        eventPublisher.publishEvent(RecallRequestEvent.of(transaction.getHold(), recallUser));
+
+      }
+    }
```

Comment:
what's the point to repeat same action multiple times here?

Reply:
after  updateTransactionOnLoanRecallRequested(transaction); I added - return

Reply:
Added return in the end of the else